### ISSUE TITLE
typeof x === undefined does not raise a type error

### ISSUE THREAD

**Issue by Jack-Works** [microsoft/TypeScript#42791]
# Bug Report

As title describes

### 🔎 Search Terms

typeof undefined

### 🕗 Version & Regression Information

Happened in nightly

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=4.3.0-dev.20210213#code/C4TwDgpg9gZgBAIgXAvGuBXAdgEwjASywhyA

### 💻 Code

```ts
typeof "" === undefined
```

### 🙁 Actual behavior

Nothing

### 🙂 Expected behavior

undefined is not comparable with string

**Comment by MartinJohns** [comment]
Duplicate of #29405 / #29200. Used search terms: `typeof undefined in:title`

**Comment by Jack-Works** [comment]
oh, I didn't search them in closed issues. let me read it
